framework/packages/abstract-client/src/client.rs

```diff
@@ -57,10 +57,18 @@ impl<Chain: CwEnv> AbstractClient<Chain> {
     ) -> AbstractClientResult<Account<Chain>> {
         Account::from_namespace(&self.abstr, namespace)
     }
+
+    pub fn chain(&self) -> Chain {
```

Comment:
We originally also had `chain` as the function name instead of `environment` but rust-std exports a trait with a `chain` function so it caused weird conflicts in some places, which is why I renamed it to `environment`. 

I think it's important to be consistent tho, so id prefer keeping this exposed in the `Infrastructure` trait exclusively.

We could rename the `environment` function to `env` or something else?

Reply:
I'd rather have the `Infrastructure` trait remain internal and only expose what we want through the client. Happy to change the name of this function though.

Reply:
In that case, where are you using this function? If you want to don't want to expose `Infrastructure` then you probably don't want to expose anything chain-related? Is that feasible?

Reply:
The trait also has other stuff like returning an instance of `Abstract` which we don't want to do. I added `chain()` here to be used in the subscription module tests for raw queries. I tried to allow a `QuerierWrapper` to be returned by the client but with the `Rc<RefCell<...>>` stuff it didn't seem possible.

Reply:
Maybe we just need to expose `chain` in `Mock` environments then?

Reply:
Raw queries is a feature that's been requested but not yet implemented: https://linear.app/abstract-sdk/issue/ORC-61/add-raw-query-support

Maybe we can split the `Infrastructure` trait into `AbstractInfrastructure` and `Environment` or whatever where 

```rust
trait AbstractInfrastructure<Chain> : Environment<Chain> {
   // construct Abstract<Chain>
}

trait Environment<Chain> {
 fn chain(&self) -> Chain
}
```

Because getting the environment is something that I've had to do quite a lot (hence there's a trait that provides it).

Reply:
Sure, I can look at adding that in a future change.